### PR TITLE
fix(material/schematics): rename references in MDC generate schematic

### PR DESCRIPTION
**Pull request by crisbeto**
Currently the `ng generate` migration that migrates from the legacy symbols to the non-legacy ones renames references, but only in some specific cases (e.g. `imports` of an `NgModule`). These changes expand the migration to cover any reference within the file. I've also tried to reduce the amount of migration data that we need to maintain.

### CODE REVIEW

File: src/material/schematics/ng-generate/mdc-migration/rules/ts-migration/runtime-migration.ts [hunk 1]
```diff
@@ -240,12 +300,48 @@ export class RuntimeCodeMigration extends Migration<ComponentMigrator[], Schemat
...
+      nodeSymbol.declarations[0] === importSymbol.declarations[0]
+    );
+  }
+
+  /**
+   * Strips "legacy" from the name of a symbol or an import.
+   * Returns null if it doesn't have a legacy name.
+   */
+  private _removeLegacy(name: string): string | null {
+    const legacyRegex = /legacy_|legacy-|legacy/i;
```
Comment (mmalerba) [inline src/material/schematics/ng-generate/mdc-migration/rules/ts-migration/runtime-migration.ts:344]: I think you can condense this to `/legacy[_-]?/i`

**Review by mmalerba — COMMENTED** [review]
Awesome work!!

File: src/material/schematics/ng-generate/mdc-migration/rules/ts-migration/runtime-migration.ts [hunk 2]
```diff
@@ -22,22 +24,125 @@ export class RuntimeCodeMigration extends Migration<ComponentMigrator[], Schemat
...
+  }
+
+  /** Finds the imported symbols in a file that need to be migrated. */
+  private _findImportsToMigrate(sourceFile: ts.SourceFile) {
+    const importSpecifiers = new Map<ts.ImportSpecifier, string>();
+    const moduleSpecifiers = new Map<ts.StringLiteral, string>();
+
+    for (const statement of sourceFile.statements) {
+      if (
+        !ts.isImportDeclaration(statement) ||
```
[the remark this answers is not in the document]
Comment (crisbeto) [reply src/material/schematics/ng-generate/mdc-migration/rules/ts-migration/runtime-migration.ts:59]: I'm not sure that's much more readable tbh. I ended up replacing the early `continue` statement with an `if` instead.

File: src/material/schematics/ng-generate/mdc-migration/rules/ts-migration/runtime-migration.ts [hunk 3]
```diff
@@ -22,22 +24,125 @@ export class RuntimeCodeMigration extends Migration<ComponentMigrator[], Schemat
...
+
+    const replacements: Replacement[] = [];
+    const walk = (node: ts.Node) => {
+      // Imports are handled separately.
+      if (ts.isImportDeclaration(node)) {
+        return;
+      }
+
+      if (ts.isIdentifier(node)) {
+        for (const [specifier, newName] of importSpecifiers.entries()) {
```
[the remark this answers is not in the document]
Comment (crisbeto) [reply src/material/schematics/ng-generate/mdc-migration/rules/ts-migration/runtime-migration.ts:101]: Maybe I'm misunderstading this, but I think that this is covered by the first early exit check inside `_isReferenceToImport`.

**Comment by crisbeto** [comment]
The feedback should be addressed.

File: src/material/schematics/ng-generate/mdc-migration/rules/ts-migration/runtime-migration.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (mmalerba) [reply src/material/schematics/ng-generate/mdc-migration/rules/ts-migration/runtime-migration.ts:101]: The way you have it now ensures that we just have to do minimal work for each element of the map, but we still have to iterate over the entire map for every node. With the change I'm proposing we would only have to iterate over a fraction of the map at each node

[the remark this answers is not in the document]
Comment (crisbeto) [reply src/material/schematics/ng-generate/mdc-migration/rules/ts-migration/runtime-migration.ts]: Done. I ended up going for a simpler `Map<string, ts.ImportSpecifier>` instead.

**Review by mmalerba — APPROVED** [review]
LGTM

**Pull request merged**